internal/resources/openshift.go

```diff
@@ -0,0 +1,675 @@
+	}
+
+	m.logger.Info("Removing OpenShift finalizer")
+	controllerutil.RemoveFinalizer(cc, openshiftFinalizer)
+	err = m.client.Update(ctx, cc)
+	if err != nil {
+		return Result{err: err}
+	}
+	return Done
+}
+
+// getOrUpdateNamespace expects operator's namespace to exist with proper
+// labels for OpenShift deployment
+func (m *OpenShiftManager) getOrUpdateNamespace(
+	ctx context.Context, nsname string) (*corev1.Namespace, bool, error) {
+	// ---
+	nsCurr, err := m.getNamespace(ctx, nsname)
+	if err != nil {
+		m.logger.Error(err, "Failed to get Namespace", "nsname", nsname)
+		return nil, false, err
+	}
+	labelKey := clusterMonitoringLabelKey
+	labelVal := "true"
+	if nsCurr.Labels[labelKey] == labelVal {
+		return nsCurr, false, nil // OK
+	}
+	nsWant := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   nsname,
+			Labels: nsCurr.Labels,
+		},
+	}
+	nsCurr.Spec.DeepCopyInto(&nsWant.Spec)
+	nsWant.Labels[labelKey] = labelVal
+	err = m.client.Update(ctx, nsWant)
+	if err != nil {
+		m.logger.Error(err, "Failed to update Namespace", "nsname", nsname)
+		return nil, false, err
+	}
+	return nsWant, true, nil
+}
+
+func (m *OpenShiftManager) getNamespace(
+	ctx context.Context, nsname string) (*corev1.Namespace, error) {
+	// ---
+	nsKey := types.NamespacedName{Name: nsname}
+	ns := &corev1.Namespace{}
+	err := m.client.Get(ctx, nsKey, ns)
+	if err != nil {
+		return nil, err
+	}
+	return ns, err
+}
+
+func (m *OpenShiftManager) getOrCreateServiceAccount(
+	ctx context.Context) (*corev1.ServiceAccount, bool, error) {
+	// ---
+	saCurr, saKey, err := m.getServiceAccount(ctx)
+	if err == nil {
+		return saCurr, false, nil // OK
+	}
+	if !errors.IsNotFound(err) {
+		m.logger.Error(err, "Failed to get ServiceAccount", "key", saKey)
+		return nil, false, err
+	}
+	saWant := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: saKey.Namespace,
+			Name:      saKey.Name,
+		},
+	}
+	err = m.client.Create(ctx, saWant, &rtclient.CreateOptions{})
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			m.logger.Info("ServiceAccount already exists", "key", saKey)
+		} else {
+			m.logger.Error(err, "Failed to create ServiceAccount",
+				"key", saKey)
+		}
+		return nil, false, err
+	}
+	return saWant, true, nil
+}
+
+func (m *OpenShiftManager) getServiceAccount(
+	ctx context.Context) (*corev1.ServiceAccount, types.NamespacedName, error) {
+	// ---
+	saKey := m.serviceAccountNsname()
+	sa := &corev1.ServiceAccount{}
+	err := m.client.Get(ctx, saKey, sa)
+	if err != nil {
+		return nil, saKey, err
+	}
+	return sa, saKey, nil
+}
+
+func (m *OpenShiftManager) serviceAccountNsname() types.NamespacedName {
+	if len(m.cfg.ServiceAccountName) > 0 {
+		return types.NamespacedName{Name: m.cfg.ServiceAccountName}
+	}
+	return types.NamespacedName{
+		Namespace: m.cfg.PodNamespace,
+		Name:      serviceAccountName,
+	}
+}
+
+func (m *OpenShiftManager) getOrCreateSccRole(
```

Comment:
This is really trivial but I might as well say it: I think that common idiom for go is to retain all capital letters for initialisms in a camel case name. So instead of "getSccRole" I think "getSCCRole" is considered more correct.  Like I said, minor, but it's something I've been trying to remember to do when I write new code.

Reply:
ACK -- will fix.